thirdeye/thirdeye-frontend/app/pods/components/filter-select/component.js

```diff
@@ -0,0 +1,181 @@
+import Ember from 'ember';
+import { task, timeout } from 'ember-concurrency';
+
+/**
+ * Parse the filter Object and return a Object of groupName
+ * containing Arrays of filter options
+ * @param {Object} filters
+ */
+const buildFilterOptions = (filters) => {
+  return Object.keys(filters).map((filterName) => {
+
+    const options = filters[filterName]
+      .filter(value => !!value)
+      .map((value) => {
+        return {
+          name: value,
+          id: `${filterName}::${value}`
+        }
+      })
+
+    return {
+      groupName: `${filterName}`,
+      options
+    }
+  })
+};
+
+/**
+ * Converts the query Params JSON into
+ * powerselect compatible groups
+ * @param {Object} filters
+ */
+const convertHashToFilters = (filters) => {
+  const filterArray = [];
+
+  Object.keys(filters).forEach((filterGroup) => {
+    const options = filters[filterGroup].map((option) => {
+      return {
+        name: option,
+        id: `${filterGroup}::${option}`
+      }
+    })
+    filterArray.push(...options);
+  })
+
+  return filterArray;
+};
+
+/**
+ * Converts the powerselect grouped filters into
+ * JSON for query Params
+ * @param {Object} selectedFilters
+ */
+const convertFiltersToHash = (selectedFilters) => {
+  const filters = selectedFilters.reduce((filterHash, filter) => {
+    const [ filterGroup, filterName ] = filter.id.split('::');
+
+    filterHash[filterGroup] = filterHash[filterGroup] || [];
+    filterHash[filterGroup].push(filterName);
+
+    return filterHash;
+  }, {})
+
+  return JSON.stringify(filters);
+};
+
+/**
+ * Helper function performing the typeahead search
+ * @param   {Object} filterOptions  All Filters
+ * @param   {String} filterToMatch  Filter name to match
+ * @param   {Number} maxNum         Maximum Number of filters to display
+ * @returns {Array}                 Array that contains the ouput of the search
+ */
+const getSearchResults = (filterOptions, filterToMatch, maxNum) => {
+  let count = 0;
+  const filters = filterOptions.reduce((foundFilters, filterOption) => {
+    if (count > maxNum) {
+      return [];
+    }
+    let options = filterOption.options.filter((el) => {
+      return el.id.toLowerCase().includes(filterToMatch.toLowerCase());
+    })
+
+    if (options.length) {
+      count += options.length;
+      foundFilters.push({
+        groupName: `${filterOption.groupName} (${options.length})`,
+        options: options
+      });
+    }
+    return foundFilters;
+  }, []);
+
+  const NoMatchMessage = count
+    ? `Too Many results found (${count})`
+    : 'No results found.'
+
+  return [filters, NoMatchMessage];
+};
+
+
+export default Ember.Component.extend({
+  // Maximum filters by filter group
+  maxNumFilters: 25,
```

Comment:
groups with more than 25 will be shown as `pageKey (688)`.
Yes, ultimately, we would want to have the expand and lazy loading, but that would potentially require our own fork of the power-select, or working with the open source project.